src/main/java/com/github/maleksandrowicz93/cqrsdemo/interfaces/BaseRepository.java

```diff
@@ -0,0 +1,15 @@
+package com.github.maleksandrowicz93.cqrsdemo.interfaces;
+
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+
+import java.util.Optional;
+
+public interface BaseRepository<T, ID> {
```

Comment:
To jeszcze można byłoby podzielić na `ReadRepository` i `WriteRepository`, ale to takie "nice to have".

Reply:
StudentRepo oraz JpaStudentRepo też rozumiem, że powinny być rozdzielone (wersje Read oraz Write)?

Reply:
Obczaj w tej prezentacji o CQRSie Krzysztofa Ślusarskiego, on tam ma to tak podzielone.